saleor/graphql/payment/resolvers.py

```diff
@@ -13,5 +21,42 @@ def resolve_payments(info, query):
     return gql_optimizer.query(queryset, info)
 
 
-def resolve_payment_client_token(gateway=None):
+def resolve_payment_client_token(gateway=None, user=None):
+    if user is not None:
```

Comment:
Is it really a problem? Resulting loc count is the same. I know it is minor issue, not worth the trouble, but maybe we can have some clarifying discussion here, cause I see some common thread in your review 😉 
It is probably the matter of programming taste, but I virtually always refactor in the opposite direction! I consider mutating *anything* a penalty in itself and try to remove it so things are more declarative and less sequenced. There is incredibly long list of reasons for that but you could just boil it down to: functional style is more future proof, easier to reason, refactor and debug.
There is also typing issue - after refactoring _token_config_ is _de facto_ union of NoneType and TokenConfig, not simple dataclass.
It is better to have one return call instead of two, obviously, not sure if it something to micro-optimize for.